File: frontend/src/types/evaluation.ts
// Evaluation data types
export interface Evaluation {
    id: string
    name: string
    courseSubject: string
    maximumMarks: number
    answerType: 'printed' | 'handwritten'
    createdAt: string
    status: 'draft' | 'in-progress' | 'completed'
    totalSheets?: number
}

// Form data for creating evaluations
export interface EvaluationFormData {
    evaluationName: string
    courseSubject: string
    maximumMarks: string
    answerType: 'printed' | 'handwritten'
}

// Evaluation status types
export type EvaluationStatus = 'not-started' | 'processing' | 'completed' | 'error'

File: backend/app/schemas/evaluation.py
"""
Evaluation management schemas for CRUD operations.
"""
from pydantic import BaseModel
from typing import Literal


class CreateEvaluationRequest(BaseModel):
    """Request schema for creating a new evaluation"""
    name: str
    subject: str
    max_marks: int
    answer_type: Literal["printed", "handwritten"] = "printed"


class EvaluationResponse(BaseModel):
    """Response schema for evaluation data"""
    id: str
    name: str
    subject: str
    max_marks: int
    answer_type: str
    status: str = "draft"


class EvaluationListResponse(BaseModel):
    """Response schema for listing evaluations"""
    success: bool = True
    evaluations: list[EvaluationResponse]


class DeleteEvaluationResponse(BaseModel):
    """Response schema for deletion"""
    success: bool = True
    deleted: str
    message: str
